lib/database_dumper.rb

```diff
@@ -1205,7 +1205,7 @@ def self.mysqldump_tsv(database, command, dest_filename)
 
   def self.mysqldump(db_name, dest_filename)
     system_pipefail!("mysqldump #{self.mysql_cli_creds} #{db_name} -r #{dest_filename} #{filter_out_mysql_warning}")
-    system_pipefail!("sed -i 's_^/\\*!50013 DEFINER.*__' #{dest_filename}")
+    system_pipefail!("ruby -i -pe '$_.gsub!(%r{^/\\*!50013 DEFINER.*\\n}, \"\")' #{dest_filename}")
```

Comment:
Are you sure the patterns are equivalent? The old `sed` pattern checks for some underscore characters which seem to have gone missing in the Ruby replacement

Reply:
The underscores characters here aren't part of the regexp, they're the delimiter for the `s` operator. Normally the syntax is `s/regexp/replacement/`, but the slashes can be replaced with other characters, as it is here `s_regexp_replacement_`. This is useful when you have slashes in the regular expression so you don't have to escape them, which must be why it's done here.

Here are some tests to demonstrate:

```sh
$ cat example-dump.sql
/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!50013 DEFINER=`db_user`@`%` SQL SECURITY DEFINER */

# The original:
$ cat example-dump.sql | sed 's_^/\\*!50013 DEFINER.*__'
/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;


# Using sed and the standard "/" delimiter:
$ cat example-dump.sql | sed 's/^\/\\*!50013 DEFINER.*//'
/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;


# The ruby version taking out the same line:
$ cat example-dump.sql  | ruby -pe '$_.gsub!(%r{^\/\*!50013 DEFINER.*\n}, "")'
/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;

# Ruby also supports using underlines as a delimiter:
$ cat example-dump.sql  | ruby -pe '$_.gsub!(%r_^/\*!50013 DEFINER.*\n_, "")'
/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
```

I do note, however, that my version does also take out the final "\n", and although that most likely won't be an issue (I wasn't even able to replicate a dump that generated a line  with `/*!50013 DEFINER.*` on it), I could put it back to make sure we're 100% like-for-like.

Reply:
Thanks for the thorough explanation! I'm happy with the example cases that you provided so we should be good :)